mentat/conversation.py

```diff
@@ -254,14 +266,42 @@ async def get_model_response(self) -> list[FileEdit]:
                     p.get("text", "") for p in prompt if p.get("type") == "text"
                 ]
                 prompt = " ".join(text_prompts)
+            max_tokens = self._get_max_tokens()
+            if max_tokens is None:
+                stream.send(
+                    f"Context size for {config.model} is not known. Please set"
+                    " maximum-context with `/config maximum_context value`.",
+                    color="light_red",
+                )
+                return []
+
+            if max_tokens - tokens < config.token_buffer:
+                if max_tokens - tokens < 0:
+                    stream.send(
+                        f"The context size is limited to {max_tokens} tokens and"
+                        f" previous messages plus system prompts use {tokens} tokens."
+                        " Please use `/clear` to reset or restart the session.",
+                        color="light_red",
+                    )
+                else:
+                    stream.send(
+                        f"The context size is limited to {max_tokens} tokens and"
+                        f" previous messages plus system prompts use {tokens} tokens,"
+                        " leaving insufficent tokens for a response. Please use"
```

Comment:
Maybe mention the token buffer here? Otherwise I can imagine this being pretty confusing if you don't know what token_buffer is

Reply:
I think I'll just remove the `/config` idea from the warning. They probably don't want to make it smaller than 1000 anyway and I don't want the warning to get too long and confusing.